aip/0231.md

```diff
@@ -0,0 +1,86 @@
+---
+aip:
+  id: 231
+  state: reviewing
+  created: 2019-06-18
+permalink: /231
+redirect_from:
+  - /0231
+---
+
+# Batch Get
+
+Some APIs need to allow users to get a specific set of resources at a consistent
+time point (e.g. using a read transaction). A batch get method provides this
+functionality.
+
+## Guidance
+
+APIs **may** support Batch Get using the following pattern:
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/shelves/-/books:batchGet"
+  };
+}
+```
+
+- The RPC's name **must** begin with `BatchGet`. The remainder of the
+  RPC name **should** be the plural form of the resource being retrieved.
+- The request and response messages **must** match the RPC name, with `-Request`
+  and `-Response` suffixes.
+- The HTTP verb **must** be `GET`.
+- The HTTP URI **must** end with `:batchGet`.
+- The URL path **should** represent the collection for the resource, matching
+  the collection used for simple CRUD operations. If the operation spans
+  parents, a dash (`-`) **may** be used as a wildcard.
+- For users with URL length restrictions (e.g. certain old browsers), they can
+  use `POST` to invoke this API method combined with the header
+  `X-HTTP-Method-Override: GET`, which instructs the server to treat the `POST`
+  as a `GET`.
+- There **must not** be a body key in the `google.api.http` annotation.
+
+### Request message
+
+The request for a batch get method **should** be specified with the following
+pattern:
+
+```proto
+message BatchGetBooksRequest {
+  // The requests specifying the books to retrieve.
+  // A maximum of 1000 books can be retreived in a batch.
+  repeated GetBookRequest requests = 1;
+}
+```
+
+- The request message **must** include one repeated field which accepts the
+  request messages specifying the resources to retrieve, as specified for
+  [standard Get methods][request-message]. The field **should** be named
+  `requests`.
+- Batch get **should not** support pagination because transactionality across
+  API calls would be extremely difficult to implement or enforce.
+- The operation **should** be atomic: it **should** fail for all resources or
```

Comment:
Since this is only a **should**, then the AIP should also provide guidance non-atomic is supported. Is the response code still 200 OK? Will there be any errors describing the partial failures?

Reply:
I think we should upgrade this to **must**.

Reply:
I will upgrade it to **must**.